CONTRIBUTION.md

```diff
@@ -0,0 +1,57 @@
+# Contribution guidlines
+## Preamble
+These guidelines are intended as "guidelines" to how an PR To Dire's Buildinggadgets should be structured. 
+Bear in mind that these are only guidelines and don't necessarily apply in each and every case, but if you choose not to follow a specific Part of this specification, you must be able to reason out why, if asked to do so.
+## PR structure
+### General Notes
+In General a PR should explain what it does, how it does this and how it was tested. It should also follow the [Code Guidelines](#code-guidelines).
+* If you are not yet finished implementing, prepend it with \[WIP]
+* A description must be provided before the PR is merged, to provide record of the Performed changes
+### Bug Fix PR's
+Your PR should include the following at least the following Information:
+* A reference to the corresponding issue or a full description of the issue being fixed
+* An explanation how this PR solves/attempts to solve the given issue
+* If testing was performed, what tests and how extensive they have been
+### Feature Request PR's
+Your PR should include the following at least the following Information:
+* A reference to the corresponding issue or a full description of the feature being implemented
+* An explanation how this PR attempts to implement the given feature
+* If testing was performed, what tests and how extensive they have been
+## Code Guidelines
+In General your code should be well-structured and easy to read. 
+We know that as of 1.12 this is not the case for great parts of the mod, but we are trying to improve upon the code-base.
+Also your PR should not contain any formatting only changes to Files, as this simply makes reviewing them unnecessarily complex.
+Try to avoid unnecessary commits in order not to pollute the commit history.
```

Comment:
I'm not sure where this line fit's best. I put it under Code guidlines, as it's about when to commit code, but it might also fit into PR Structure